lib/output.js

```diff
@@ -1185,6 +1186,35 @@ function OutputStream(options) {
         output.print(self.getValue());
     });
     DEFPRINT(AST_String, function(self, output){
+        var parent = output.parent(1);
+        // Check if there is a need to break directive by inserting an additional semicolon
+        if (parent instanceof AST_Scope) {
+            var print_semicolon = false;
+            // Check if string is the first in body or after directive
+            for (var i = 0; i < parent.body.length; i++) {
+                if (parent.body[i] instanceof AST_SimpleStatement && parent.body[i].body instanceof AST_String) {
+                    print_semicolon = parent.body[i].body === self;
+                    break;
+                }
+                if (!(parent.body[i] instanceof AST_Directive)) {
+                    break; // Breaking directive prologue
+                }
+            }
+            if (print_semicolon) {
+                var str = output.encode_string(self.getValue(), self.quote);
+                // Only output semicolons if string might be a valid directive
+                if (str.indexOf("\\") === -1) {
+                    var last = output.last();
+                    // Extra semicolon might be needed to break directive prologue
+                    if (last !== null && /[;{]\s*$/.test(last)) {
+                        output.force_semicolon();
+                    }
+                    output.force_semicolon();
+                }
+                output.print(str);
+                return;
+            }
+        }
```

Comment:
At the very least all the logic above could be refactored to simply set a flag to emit a semicolon at the end of the function and unconditionally output the string in one place. No need for encode_string to be there.

Instead of all this logic in DEFPRINT(AST_String), perhaps parser.js can just set a  property in either the last AST_Directive or the first non-directive AST_Statement or AST_String to indicate an extra semicolon should be prepended in the output. Or add an initial tree traversal pass in output.js to set such a property instead of repeating the logic for each and every AST_String.